mozilla/foundation.mozilla.org
# Update field owners

Closes #5371
Related PRs/issues #5372

This moves all the "general only" fields from `BaseProduct` to `GeneralProduct`. as per https://docs.google.com/spreadsheets/d/1cOOzLhkoICFIPTUmxQnrH10HZz1qq5eHF51qgj1p8QM/edit#gid=0, with a custom migration to ensure that even though Django does this by removing the fields from `BaseProduct` and then building brand new fields with the same name on `GeneralProduct`, we cache the values before that destructive relocation happens, and then restore the values for all those fields once the fields have been rebuilt.

The "easiest" way to test this for devs is:

1. check out pni-v4-2020
1. run `inv new-db`
1. run `docker-compose up` and navigate to http://localhost:8000/cms/buyersguide/generalproduct/edit/1/
1. set every choice to "yes" or "true" or checked, and set every textfield you can to "this is a custom test string".
1. save that product.
1. exit docker
1. check out `update-field-owners`
1. run `inv migrate`, which should apply the fancy migration
1. run `docker-compose up`
1. load up http://localhost:8000/cms/buyersguide/generalproduct/edit/1/ again
1. look at every field you updated.

If everything went well, everything should still be yes/true/checked/custom string.

For the design/pm side of things. the important part is that the review app shows the fields that should be visible for General Products as being there in the CMS, while not showing up in the Software Product part of the CMS.
